app/controllers/api/v1/users_controller.rb

```diff
@@ -27,6 +27,19 @@ def show
       end
     end
 
+    def search
+      if params[:q].present?
+        users = User.where("name ~* ?", params[:q])
+      else
+        users = User.none
+      end
+      users = users.by_email
```

Comment:
Any reason to order the response by email rather than name?
I'd expect a name ordering seeing as that is what we are searching on ...